bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/hbasesnapshots/SnapshotUtils.java

```diff
@@ -0,0 +1,293 @@
+      List<ImportConfig.SnapshotInfo> snapshotdetails,
+      Map<String, String> configurations,
+      String projectId,
+      String sourcePath,
+      String restorePath) {
+
+    return snapshotdetails.stream()
+        .map(
+            snapshotInfo ->
+                SnapshotConfig.builder()
+                    .setProjectId(projectId)
+                    .setSourceLocation(sourcePath)
+                    .setRestoreLocation(restorePath)
+                    .setSnapshotName(snapshotInfo.getSnapshotName())
+                    .setTableName(snapshotInfo.getbigtableTableName())
+                    .setConfigurationDetails(configurations)
+                    .build())
+        .collect(Collectors.toList());
+  }
+
+  public static BackOff createBackOff(
+      long backoffInitialIntervalInMillis, long backoffMaxIntervalInMillis, int maxRetries) {
+    return FluentBackoff.DEFAULT
+        .withInitialBackoff(Duration.millis(backoffInitialIntervalInMillis))
+        .withMaxRetries(maxRetries)
+        .withMaxBackoff(Duration.millis(backoffMaxIntervalInMillis))
+        .backoff();
+  }
+
+  /**
+   * Creates restore path based on the input configuration
+   *
+   * @param importConfig - Job Configuration
+   */
+  public static void setRestorePath(ImportConfig importConfig) {
+    importConfig.setRestorepath(
+        formatRestorePath(importConfig.getRestorepath(), importConfig.getSourcepath()));
+  }
+
+  /**
+   * Parses the provided input to generate snapshot names and corresponding bigtable names. For
+   * single snapshot names the following are valid formats: If both Snapshotname and bigtablename
+   * are same then only snapshotname can be provided If bigtablename is different then should be
+   * provided in the format snapshotname:bigtablename Multiple snapshots can be provided in
+   * snapshot1:table1,snapshot2:table2 format or snapshot1,snapshot2 format
+   *
+   * @param snapshotNames - Snapshot names and corresponding bigtable table names.
+   */
+  public static Map<String, String> getSnapshotsFromString(String snapshotNames) {
+    Map<String, String> snapshots = new HashMap<>();
+    for (String snapshotInfo : snapshotNames.split(",")) {
+      String[] snapshotWithTableName = snapshotInfo.split(":");
+      if (snapshotWithTableName.length == 2)
+        snapshots.put(snapshotWithTableName[0], snapshotWithTableName[1]);
+      else if (snapshotWithTableName.length == 1)
+        snapshots.put(snapshotWithTableName[0], snapshotWithTableName[0]);
+      else
+        throw new IllegalArgumentException(
+            "Invalid specification format for snapshots. Expected format is snapshot1:table1,snapshot2:table2");
+    }
+    return snapshots;
+  }
+
+  public static String formatRestorePath(String providedPath, String hbaseSnapshotsPath) {
+    return providedPath == null
+        ? SnapshotUtils.getNamedDirectory(hbaseSnapshotsPath, "restore")
+        : SnapshotUtils.appendCurrentTimestamp(providedPath);
```

Comment:
this should be more unique to avoid collisions (rather than minute timestamp, it should use. a uuid)